### PR TITLE
Support for language bindings (Python 3 and Ch Professional implemented), WIP C++ header-only wrapper implementation, CI/CD, test rig, build, and documentation improvements.

### PR DESCRIPTION
**Pull request by johnsonjh**
* Clean up some warnings when building on AIX with XL C/C++ 16
* Add bindings for Ch Professional and Python 3
* Add sirtests++ test rig for testing the C++ wrapper
* Improve CI/CD; add jobs for Semgrep scans, C++ wrapper, Ch bindings, Python bindings, etc.
* Update Makefiles and documentation

### CODE REVIEW

**Comment by johnsonjh** [comment]
[Coverage](https://github.com/aremmell/libsir/pull/292#issuecomment-1712929363)

**Comment by aremmell** [comment]
Hrm, coverage on sirinternal.c is no bueno.

File: include/sir/queue.h [hunk 1]
```diff
@@ -54,4 +54,3 @@ bool _sir_queue_push(sir_queue* q, void* data);
 bool _sir_queue_pop(sir_queue* q, void** data);
 
 #endif /* !_SIR_QUEUE_H_INCLUDED */
-
```
Comment (aremmell) [inline include/sir/queue.h:57]: Removed trailing newline? I'l fix.

Comment (aremmell) [reply include/sir/queue.h:57]: Weird, there is a line 57 that's empty in my file...

**Review by aremmell — COMMENTED** [review]
Will wait for you to get back to discuss comments. Otherwise ready to merge.

**Review by johnsonjh — COMMENTED** [review]
👍🏻

**Pull request merged**